src/snowflake/snowpark/stored_procedure.py

```diff
@@ -771,7 +771,8 @@ def _do_register_sp(
             if package_name not in self._session._packages:
                 packages = list(self._session._packages.values()) + [this_package]
         else:
-            if not any(package_name in p for p in packages):
+            package_names = [p if isinstance(p, str) else p.__name__ for p in packages]
+            if not any(p.startswith(package_name) for p in package_names):
```

Comment:
Is this because `package_names` contain multiple entries of the same package with different versions?

Reply:
good question, I tried to mimick the code above `package_name in p` but though that packages will be given as strings in the form `pandas==2.0.3`. Hence, comparing with startswith seems the right idea. `in` seems wrong, because what if there's a package `connector` requested? that might conflict with `snowflake-connector` or `connector-java`. Prob. `startswith` is not even correct - do you know if there are some existing util functions to deal with correct resolution? Potentially splitting the string after `==, !=, <=, ...` into two parts and then comparing the first part may be the right thing to do. Wdyt?